tap_postgres/sync_strategies/logical_replication.py

```diff
@@ -288,15 +364,35 @@ def consume_message(streams, state, msg, time_extracted, conn_info, end_lsn):
             raise Exception("unrecognized replication operation: {}".format(c['kind']))
 
 
-        singer.write_message(record_message)
+        yield record_message
         state = singer.write_bookmark(state,
                                       target_stream['tap_stream_id'],
                                       'lsn',
                                       lsn)
+
+
+def consume_message(streams, state, msg, time_extracted, conn_info, end_lsn, message_format="1"):
+    payload = json.loads(msg.payload)
+    lsn = msg.data_start
+
+    streams_lookup = {}
+    for s in streams:
+        streams_lookup[s['tap_stream_id']] = s
```

Comment:
This looks like it could be simplified as a dict comprehension?

Reply:
Yup, it could be. That was copied over from the original implementation. Will do :+1: